src/polycap-description.c

```diff
@@ -0,0 +1,455 @@
+
+	//allocate profile memory
+	description->profile = malloc(sizeof(polycap_profile));
+	if(description->profile == NULL){
+		printf("Could not allocate description->profile memory.\n");
+		exit(1);
+	}
+	description->profile->z = malloc(sizeof(double)*(profile->nmax+1));
+	if(description->profile->z == NULL){
+		printf("Could not allocate description->profile->z memory.\n");
+		exit(1);
+	}
+	description->profile->cap = malloc(sizeof(double)*(profile->nmax+1));
+	if(description->profile->cap == NULL){
+		printf("Could not allocate description->profile->cap memory.\n");
+		exit(1);
+	}
+	description->profile->ext = malloc(sizeof(double)*(profile->nmax+1));
+	if(description->profile->ext == NULL){
+		printf("Could not allocate description->profile->ext memory.\n");
+		exit(1);
+	}
+	
+	// copy description->profile values to profile
+	for(i=0;i<=profile->nmax;i++){
+		description->profile->z[i] = profile->z[i];
+		description->profile->cap[i] = profile->cap[i];
+		description->profile->ext[i] = profile->ext[i];
+	}
+
+	// Calculate open area
+	description->open_area = (description->profile->cap[0]/description->profile->ext[0]) * (description->profile->cap[0]/description->profile->ext[0]) * description->n_cap;
+
+	return description;
+}
+
+//===========================================
+// get the polycap_profile from a polycap_description
+polycap_profile* polycap_description_get_profile(polycap_description *description)
+{
+	polycap_profile *profile;
+	int i;
+
+	//allocate profile memory
+	profile = malloc(sizeof(polycap_profile));
+	if(profile == NULL){
+		printf("Could not allocate profile memory.\n");
+		exit(1);
+	}
+	profile->nmax = description->profile->nmax;
+	profile->z = malloc(sizeof(double)*(profile->nmax+1));
+	if(profile->z == NULL){
+		printf("Could not allocate profile->z memory.\n");
+		exit(1);
+	}
+	profile->cap = malloc(sizeof(double)*(profile->nmax+1));
+	if(profile->cap == NULL){
+		printf("Could not allocate profile->cap memory.\n");
+		exit(1);
+	}
+	profile->ext = malloc(sizeof(double)*(profile->nmax+1));
+	if(profile->ext == NULL){
+		printf("Could not allocate profile->ext memory.\n");
+		exit(1);
+	}
+
+	// copy description->profile values to profile
+	for(i=0;i<=profile->nmax;i++){
+		profile->z[i] = description->profile->z[i];
+		profile->cap[i] = description->profile->cap[i];
+		profile->ext[i] = description->profile->ext[i];
+	}
+
+	return profile;
+
+}
+
+//===========================================
+// for a given array of energies, and a full polycap_description, get the transmission efficiencies.
+//   NOTE:
+// -Does not make photon image arrays (yet)
+// -in polycap-capil.c some leak and absorb counters are commented out (currently not monitored)
+int polycap_description_get_transmission_efficiencies(polycap_description *description, size_t n_energies, double *energies, double **efficiencies)
+{
+//why return an integer? Something like 0 for succes, -1 for fail, ...?
```

Comment:
Yep, 1 voor succes, 0 voor een foutmelding. Later kunnen we eventueel eens denken aan betere error handling met [GError](https://developer.gnome.org/glib/stable/glib-Error-Reporting.html) of zoiets.